.github/actions/setup-node/action.yml

```diff
@@ -0,0 +1,97 @@
+    required: false
+    default: false
+    description: >
+      Set this option if you want the action to check for the latest available version that satisfies the
+      version spec.
+  registry-url:
+    required: false
+    description: >
+      Optional registry to set up for auth. Will set the registry in a project level .npmrc and .yarnrc file,
+      and set up auth to read in from env.NODE_AUTH_TOKEN.
+  scope:
+    required: false
+    description: >
+      Optional scope for authenticating against scoped registries. Will fall back to the repository owner
+      when using the GitHub Packages registry (https://npm.pkg.github.com/).
+  token:
+    required: false
+    description: >
+      Used to pull node distributions from node-versions. Since there's a default, this is typically not
+      supplied by the user. When running this action on github.com, the default value is sufficient.
+      When running on GHES, you can pass a personal access token for github.com if you are experiencing
+      rate limiting.
+    default: ${{ github.server_url == 'https://github.com' && github.token || '' }}
+  cache:
+    required: false
+    description: >
+      Used to specify a package manager for caching in the default directory. Supported values: npm, yarn, pnpm.
+  cache-dependency-path:
+    required: false
+    description: >
+      Used to specify the path to a dependency file: package-lock.json, yarn.lock, etc.
+      Supports wildcards or a list of file names for caching multiple dependencies.
+  always-auth:
+    required: false
+    default: false
+    description: >
+      Set always-auth in npmrc.
+  mirror:
+    required: false
+    description: >
+      Used to specify an alternative mirror to download Node.js binaries from.
+  mirror-token:
+    required: false
+    description: >
+      The token used as Authorization header when fetching from the mirror.
+
+runs:
+  using: composite
+  steps:
+    - name: Setup Scripts
+      shell: pwsh
+      run: |
+        (Resolve-Path '${{ github.action_path }}/../../../Scripts').Path >> $Env:GITHUB_PATH
+
+    - name: Set up Node.js
+      uses: actions/setup-node@49933ea5288caeca8642d1e84afbd3f7d6820020 # v4.4.0
+      with:
+        node-version: ${{ inputs.node-version }}
+        node-version-file: ${{ inputs.node-version-file }}
+        architecture: ${{ inputs.architecture }}
+        check-latest: ${{ inputs.check-latest }}
+        registry-url: ${{ inputs.registry-url }}
+        scope: ${{ inputs.scope }}
+        token: ${{ inputs.token }}
+        cache: ${{ inputs.cache }}
+        cache-dependency-path: ${{ inputs.cache-dependency-path }}
+        always-auth: ${{ inputs.always-auth }}
+        mirror: ${{ inputs.mirror }}
+        mirror-token: ${{ inputs.mirror-token }}
+
+    - name: Save Node.js Version Into Environment Variable
+      shell: pwsh
+      run: |
+        $nodeVersion = node --version
+        Set-GitHubEnv 'NODE_VERSION_TO_USE' $nodeVersion
```

Comment:
I was also thinking of printing this value out, so we can see it from the workflow run. However, the user can still figure it out, I guess: 
![image](https://github.com/user-attachments/assets/3a2e5092-11ef-4933-b12f-a0d62f3c6434)

Do you think that's needed?

Reply:
Yeah, that would be nice, adding it.